kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/AppendRecordsContext.java

```diff
@@ -51,21 +51,21 @@ public static AppendRecordsContext get(final KafkaTopicManager topicManager,
                                            final Map<TopicPartition, PendingTopicFutures> pendingTopicFuturesMap,
                                            final ChannelHandlerContext ctx) {
         AppendRecordsContext context = RECYCLER.get();
-        context.topicManager = topicManager;
         context.startSendOperationForThrottling = startSendOperationForThrottling;
         context.completeSendOperationForThrottling = completeSendOperationForThrottling;
         context.pendingTopicFuturesMap = pendingTopicFuturesMap;
+        context.topicManager = topicManager;
         context.ctx = ctx;
 
         return context;
     }
 
     public void recycle() {
```

Comment:
I think they are removed recycle method in this commit: https://github.com/datastax/starlight-for-kafka/commit/0b6fd77c0759c54d92ac5aee9b585d1c71c88712

Reply:
Good call, then it's probably not needed any more